src/engine.rs

```diff
@@ -126,25 +146,46 @@ impl Engine {
         }
     }
 
+    pub fn add_filter_list<'a>(&'a mut self, filter_list: &str) -> &'a mut Engine {
+        let rules = filter_list.lines().map(str::to_string).collect::<Vec<_>>();
+        let (parsed_network_filters, parsed_cosmetic_filters) = parse_filters(&rules, true, true, true);
+
+        for rule in parsed_network_filters {
+            self.add_network_filter(rule);
+        }
+
+        for rule in parsed_cosmetic_filters {
+            self.add_cosmetic_filter(rule);
+        }
+
+        self
+    }
+
     pub fn filter_add<'a>(&'a mut self, filter: &str) -> &'a mut Engine {
-        let filter_parsed = NetworkFilter::parse(filter, true);
-        match filter_parsed
-        .map_err(BlockerError::from)
-        .and_then(|f| self.blocker.filter_add(f)) {
-            Ok(_b) => self,
-            Err(e) => {
-                match e {
-                    BlockerError::BlockerFilterError(e) => eprintln!("Encountered filter error {:?} when adding", e),
-                    BlockerError::BadFilterAddUnsupported => eprintln!("Adding filters with `badfilter` option dynamically is not supported"),
-                    BlockerError::FilterExists => eprintln!("Filter already exists"),
-                    e => eprintln!("Encountered unexpected error {:?} when checking for filter existence", e),
-                }
-                
-                self
-            }
+        let filter_parsed = parse_filter(filter, true, true, true);
+        match filter_parsed {
+            Ok(ParsedFilter::Network(filter)) => self.add_network_filter(filter),
+            Ok(ParsedFilter::Cosmetic(filter)) => self.add_cosmetic_filter(filter),
+            _ => {},
```

Comment:
nit: when doing network filters, there was a wish to have some debug output in case provided input is bad, here cosmetic filter errors get swallowed completely